frontend/src/components/sidebar/index.tsx

```diff
@@ -7,16 +7,32 @@ import MenuIcon from '@material-ui/icons/Menu';
 import { useWindowWidth } from '@react-hook/window-size/throttled';
 import clsx from 'clsx';
 import React, { useState } from 'react';
-import { NavLink } from 'react-router-dom';
 
-import { WINDOW_WIDTH_THRESHOLD } from '../../constants/global';
-import {
-  SidebarOption,
-  SideberButtonConfig,
-} from '../../constants/sidebarOptions';
+import { RoleType, SidebarOption } from '../../model';
 import sidebarStyles from './sidebar.module.scss';
+import SidebarButton from './SidebarButton';
 
 const sidebarWidth = 200;
+const WINDOW_WIDTH_THRESHOLD = 780;
+
+const sidebarOptionsDefault = [
+  { name: 'Summary', endpoint: 'summary' },
+  { name: 'Rota', endpoint: 'rota' },
+];
+
+const sidebarOptionsManager = [
+  ...sidebarOptionsDefault,
+  { name: 'Employees', endpoint: 'employees' },
+];
+
+const getSidebarOptions = (role: RoleType) => {
+  switch (role) {
+    case RoleType.Manager:
+      return [...sidebarOptionsManager];
```

Comment:
Is using `...sidebarOptionsManager` the same as using `Object.assign`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Basically `[...array]` creates a copy of the array (not a deep copy though).

Reply:
https://stackoverflow.com/a/48582976

The usage of spread operator here is to create new array with values of `sidebarOptionsManager` array (meaning mutations to the new array do not change the `sidebarOptionsManager` array).

I think you want to know difference between `Object.assign` and spread operator to create/mutate an object? Typically I would only use Object.assign if I need to mutate an object (to keep target object's prototype) rather than create a new instance.

@tjanowsk Agreed that we don't need to create the new array in this case probably. IMO using `Object.freeze` is not necessary the majority of the time. Only when you must enforce immutability. I guess from container perspective (this component) - it doesn't care how the child component uses this array.